src/renderers/dom/client/ReactInputSelection.js

```diff
@@ -31,8 +31,8 @@ var ReactInputSelection = {
 
   hasSelectionCapabilities: function(elem) {
     return elem && (
-      (elem.nodeName === 'INPUT' && elem.type === 'text') ||
-      elem.nodeName === 'TEXTAREA' ||
+      (elem.nodeName.toUpperCase() === 'INPUT' && elem.type === 'text') ||
```

Comment:
Might be worth just saving the value of `nodeName.toUpperCase()` so we only call it once (same in the last 2 case below)